### PR TITLE
feat(consensus): Hardfork Transaction Builders

### PR DESCRIPTION
**Pull request by refcell**
### Description

Ports over OP Stack hardfork transaction builders from [`kona`](https://github.com/ethereum-optimism/kona) to `op-alloy`. This is part of a larger refactor discussed in discord, moving kona primitives into op-alloy to separate types from higher-level libraries.

See [fjord tx builder](https://github.com/ethereum-optimism/kona/blob/main/crates/primitives/src/fjord.rs) and [ecotone tx builder](https://github.com/ethereum-optimism/kona/blob/main/crates/primitives/src/ecotone.rs).

### CODE REVIEW

File: crates/consensus/src/hardforks/ecotone.rs [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    intent: String::from("Ecotone: beacon block roots contract deployment"),
+});
+
+/// Builder wrapper for the Ecotone network updgrade.
+#[derive(Debug, Default, Clone, Copy)]
+pub struct EcotoneTransactionBuilder;
+
+impl EcotoneTransactionBuilder {
+    /// Constructs the Ecotone network upgrade transactions.
+    pub fn build_txs() -> Vec<Bytes> {
```
Comment (mattsse) [inline crates/consensus/src/hardforks/ecotone.rs:68]: will there be more functions on this type?

if not I'd like to keep this function here but remove the `EcotoneTransactionBuilder;` type and instead introduce a hardfork module level type that exposes this like `fn ecotone_txs() + fn fjord_txs`

**Review by mattsse — APPROVED** [review]
some suggestions

File: crates/consensus/src/hardforks/ecotone.rs (hunk 1 above)
Comment (refcell) [reply crates/consensus/src/hardforks/ecotone.rs:68]: Ah that's a good idea - no there shouldn't ever be more txs on this so will take a stab at your suggestion

File: crates/consensus/src/hardforks/utils.rs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+//! Utilities for Upgrade Deposit Transactions.
+
+use alloy_primitives::{Address, Bytes};
+
+/// UpgradeTo Function 4Byte Signature
```
[the remark this answers is not in the document]
Comment (refcell) [reply crates/consensus/src/hardforks/utils.rs]: I don't think we can use `hex!` for a constant though right? Not finding a nice way to do this

[the remark this answers is not in the document]
Comment (refcell) [reply crates/consensus/src/hardforks/utils.rs]: I'll put up an issue for this to unblock :)

**Pull request merged**